scipy/signal/_spectral_py.py

```diff
@@ -1987,16 +1987,15 @@ def _fft_helper(x, win, detrend_func, nperseg, noverlap, nfft, sides):
 
     .. versionadded:: 0.16.0
     """
-    # Created strided array of data segments
+    # Created sliding window view of array
     if nperseg == 1 and noverlap == 0:
         result = x[..., np.newaxis]
     else:
-        # https://stackoverflow.com/a/5568169
         step = nperseg - noverlap
-        shape = x.shape[:-1]+((x.shape[-1]-noverlap)//step, nperseg)
-        strides = x.strides[:-1]+(step*x.strides[-1], x.strides[-1])
-        result = np.lib.stride_tricks.as_strided(x, shape=shape,
-                                                 strides=strides)
+        result = np.lib.stride_tricks.sliding_window_view(
+            x, window_shape=nperseg, axis=-1, writeable=True
+        )
+        result = result[..., 0::step, :]
```

Comment:
Also of note is that `as_strided` causes a bunch of problems for array API stuff--for example, see my old shims on this file at: https://github.com/scipy/scipy/pull/19001/files#diff-d79a2488fff0b464db411eddf05248aca4d06df9d1b34cef99f1818d5afb5ec6R2049

It looks like `sliding_window_view` is present in CuPy now, but `torch` only has `as_strided` from the version of docs I just checked. Didn't check their development branch. Not sure if that's something we'd block on, as some kind of work around/shim would be needed in any case I suppose.